src/Makefile

```diff
@@ -318,9 +318,9 @@ REDIS_SERVER_NAME=redis-server$(PROG_SUFFIX)
 REDIS_SENTINEL_NAME=redis-sentinel$(PROG_SUFFIX)
 REDIS_SERVER_OBJ=adlist.o quicklist.o ae.o anet.o dict.o server.o sds.o zmalloc.o lzf_c.o lzf_d.o pqsort.o zipmap.o sha1.o ziplist.o release.o networking.o util.o object.o db.o replication.o rdb.o t_string.o t_list.o t_set.o t_zset.o t_hash.o config.o aof.o pubsub.o multi.o debug.o sort.o intset.o syncio.o cluster.o crc16.o endianconv.o slowlog.o eval.o bio.o rio.o rand.o memtest.o syscheck.o crcspeed.o crc64.o bitops.o sentinel.o notify.o setproctitle.o blocked.o hyperloglog.o latency.o sparkline.o redis-check-rdb.o redis-check-aof.o geo.o lazyfree.o module.o evict.o expire.o geohash.o geohash_helper.o childinfo.o defrag.o siphash.o rax.o t_stream.o listpack.o localtime.o lolwut.o lolwut5.o lolwut6.o acl.o tracking.o connection.o tls.o sha256.o timeout.o setcpuaffinity.o monotonic.o mt19937-64.o resp_parser.o call_reply.o script_lua.o script.o functions.o function_lua.o commands.o
 REDIS_CLI_NAME=redis-cli$(PROG_SUFFIX)
-REDIS_CLI_OBJ=anet.o adlist.o dict.o redis-cli.o zmalloc.o release.o ae.o redisassert.o crcspeed.o crc64.o siphash.o crc16.o monotonic.o cli_common.o mt19937-64.o
+REDIS_CLI_OBJ=anet.o adlist.o dict.o redis-cli.o zmalloc.o release.o ae.o redisassert.o crcspeed.o crc64.o siphash.o crc16.o monotonic.o cli_common.o mt19937-64.o util.o sha256.o sds.o
 REDIS_BENCHMARK_NAME=redis-benchmark$(PROG_SUFFIX)
-REDIS_BENCHMARK_OBJ=ae.o anet.o redis-benchmark.o adlist.o dict.o zmalloc.o redisassert.o release.o crcspeed.o crc64.o siphash.o crc16.o monotonic.o cli_common.o mt19937-64.o
+REDIS_BENCHMARK_OBJ=ae.o anet.o redis-benchmark.o adlist.o dict.o zmalloc.o redisassert.o release.o crcspeed.o crc64.o siphash.o crc16.o monotonic.o cli_common.o mt19937-64.o util.o sha256.o sds.o
```

Comment:
@oranagra I fully understand. my first instinct was to introduce the strlcpy and strlcat in separate files, but we thought this is an overhead - but still an option.

Reply:
i see the use of sds only recently creeped into util.c.
i see 3 relatively easy ways out of this.
1. move these new functions into a new file
2. move the few path handling utilities that require sds into a new file.
3. remove the inclusion of sds.h from the header and instead add an ifdef preventing it's use if `__SDS_H` isn't present.

i don't like [2] since it moves around existing code.
i'm ok with [1] and [3], i think i prefer [3], although slightly ugly.

Reply:
> i see the use of sds only recently creeped into util.c. i see 3 relatively easy ways out of this.
> 
> 1. move these new functions into a new file
> 2. move the few path handling utilities that require sds into a new file.
> 3. remove the inclusion of sds.h from the header and instead add an ifdef preventing it's use if `__SDS_H` isn't present.
> 
> i don't like [2] since it moves around existing code. i'm ok with [1] and [3], i think i prefer [3], although slightly ugly.

I prefer moving to dedicated files rather than making some confusing define dependency. let me try that.